app/uk/gov/hmrc/gform/validation/ValidationResult.scala

```diff
@@ -29,6 +29,11 @@ class ValidationResult(
 
   val isFormValid: Boolean = lookup.values.view.forall(_.isOk)
 
+  val errorsFieldIds: List[FormComponentId] = lookup.collect {
+    case (fcId, _: FieldError)       => fcId
+    case (fcId, _: FieldGlobalError) => fcId
```

Comment:
This needs to handle also `ComponentField` which can include errors as well. Use `Date`, `Address` etc. for testing.